src/test/java/br/com/zup/library/usuario/UsuarioControllerTest.java

```diff
@@ -0,0 +1,114 @@
+package br.com.zup.library.usuario;
+
+import br.com.zup.library.utils.TestUtils;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import javax.transaction.Transactional;
+import java.time.LocalDate;
+
+import static br.com.zup.library.usuario.TipoUsuario.PADRAO;
+import static br.com.zup.library.usuario.TipoUsuario.PESQUISADOR;
+import static br.com.zup.library.utils.TestFactory.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+@ActiveProfiles("test")
+class UsuarioControllerTest {
+
+    public final String URI_USUARIOS = "/api/v1/usuarios";
+    @Autowired
+    private TestUtils testUtils;
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private UsuarioRepository usuarioRepository;
+
+    @Test
+    @DisplayName("deve cadastrar um novo Usuario do tipo PADRAO e retornar 200 e o id do Usuario")
+    void deveCadastrarUsuarioTipoPadrao() throws Exception {
+        var request = criaNovoUsuarioPadrao();
+
+        var response = mockMvc.perform(testUtils.montaRequisicaoPost(request, URI_USUARIOS))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        var resposta = testUtils.fromJson(response, NovoUsuarioResponse.class);
+
+        usuarioRepository.findById(resposta.getId())
+                .ifPresent(usuario -> assertAll(
```

Comment:
o que acontece se o usuário nao for encontrado? acredito que seu teste vai passar com sucesso

Reply:
resolvido